Simulator/Recorders/XmlRecorder.h

```diff
@@ -57,16 +56,34 @@ class XmlRecorder : public IRecorder {
    ///  Registered to OperationManager as Operation::printParameters
    virtual void printParameters() override;
 
+   /// Store the neuron number and all the events for this neuron that registered in the variable owner class
+   void registerVariables(string varName, EventBuffer &recordVar) override;
+
 protected:
+   // a signle neuron number
+   string single_neuron_name;
+
+   // all events for a single neuron
+   EventBuffer *variable_first;
```

Comment:
These don't meet our requirements for member variable name styles. Also, needs more comments. I know these are just stepping stones along the way, but good to get in the habit, and most of the commenting will be reusable.